src/main/java/redis/clients/jedis/JedisClusterInfoCache.java

```diff
@@ -86,21 +86,24 @@ public void discoverClusterSlots(Jedis jedis) {
       rediscovering = true;
 
       try {
-        this.slots.clear();
-
         List<Object> slots = jedis.clusterSlots();
 
+        // We should clear slots after getting result about cluster slots, because if we got exception or timeout in
```

Comment:
Got you. And I'm not sure what is better... It could get right node and don't stuck at all. In worst case - we will get same result with rediscovering call. I would prefer to give a chance and think that more slots is still on place. But if you think that it's better to publish clear state - can revert this part.